packages/dashboard/src/components/segmentsTable.tsx

```diff
@@ -0,0 +1,231 @@
+        "segmentDownloadRequest",
+        "setSegmentDownloadRequest",
+        "apiBase",
+        "workspace",
+      ]),
+    );
+  const segments =
+    segmentsRequest.type === CompletionStatus.Successful
+      ? segmentsRequest.value
+      : [];
+  const workspace =
+    workspaceRequest.type === CompletionStatus.Successful
+      ? workspaceRequest.value
+      : null;
+
+  const segmentsRow: Row[] = [];
+
+  segments.forEach((segment) => {
+    const row: Row = {
+      id: segment.id,
+      name: segment.name,
+      updatedAt: segment.updatedAt
+        ? new Date(segment.updatedAt).toISOString()
+        : "Not Updated",
+      journeys: segment.journeys ?? "",
+      lastRecomputed: segment.lastRecomputed
+        ? new Date(segment.lastRecomputed).toISOString()
+        : "Not Re-Computed ",
+    };
+    segmentsRow.push(row);
+  });
+
+  const {
+    setSegmentDeleteRequest,
+    apiBase,
+    segmentDeleteRequest,
+    deleteSegment,
+  } = useAppStore((store) =>
+    pick(store, [
+      "setSegmentDeleteRequest",
+      "apiBase",
+      "segmentDeleteRequest",
+      "deleteSegment",
+    ]),
+  );
+
+  const setDeleteResponse = (
+    _response: EmptyResponse,
+    deleteRequest?: DeleteSegmentRequest,
+  ) => {
+    if (!deleteRequest) {
+      return;
+    }
+    deleteSegment(deleteRequest.id);
+  };
+
+  if (!workspace) {
+    console.error("No workspace found");
+    return null;
+  }
+
+  return (
+    <DataGrid
+      rows={segmentsRow}
+      sx={{
+        height: "100%",
+        width: "100%",
+        // disable cell selection style
+        ".MuiDataGrid-cell:focus": {
+          outline: "none",
+        },
+        // pointer cursor on ALL rows
+        "& .MuiDataGrid-row:hover": {
+          cursor: "pointer",
+        },
+      }}
+      getRowId={(row) => row.id}
+      onRowClick={(params) => {
+        router.push({
+          pathname: `/segments/${params.id}`,
+        });
+      }}
+      autoPageSize
+      columns={[
+        {
+          field: "name",
+          headerName: "Name",
+        },
+        {
+          field: "updatedAt",
+          headerName: "Updated At",
+        },
+        {
+          field: "journeys",
+          headerName: "Journeys used by",
+          renderCell: ({ row }: { row: Row }) => {
+            const currentRow = row;
+            if (currentRow.journeys === "No Journey") {
+              return (
+                <div>
+                  <p>No Journey</p>
+                </div>
+              );
+            }
+            const journeys = currentRow.journeys.split(",");
+            return (
+              <div>
+                {journeys.map((journey) => {
+                  const [journeyName, journeyId] = journey.split("|");
+                  return (
+                    <Button
+                      key={journey}
+                      onClick={(e) => {
+                        e.stopPropagation();
+                        router.push({
```

Comment:
A few things need here:

1. Rather than a button that uses the router to navigate, can you use a nextjs link?
2. Can you style the link so that it has max character limit for journeys with long names, with a "..." appended to the end, and wrapped in a `<Tooltip>` with the full name.
3. Can you style the link so that it has a light grey background color, so that it's clearly distinguishable from neighboring links?

This feedback is applicable to the other page's related resources columns as well.